src/cli/parser.rs

```diff
@@ -1,89 +1,34 @@
+use crate::cli::args::DownloadArgs;
+use crate::cli::args::HistoryArgs;
+use crate::cli::args::ResumeArgs;
 use crate::core::config::Config;
 use crate::core::engine::Engine;
 use crate::core::errors::RawstErr;
 use crate::core::history::HistoryManager;
-use crate::core::io::{config_exist, get_cache_sizes, read_links};
+use crate::core::io::{get_cache_sizes, read_links};
 use crate::core::task::HttpTask;
 
 use std::sync::atomic::Ordering;
 
 use base64::{prelude::BASE64_STANDARD, Engine as Base64Engine};
 use chrono::prelude::Local;
 
-use clap::{crate_authors, crate_description, crate_name, crate_version, value_parser};
-use clap::{Arg, ArgMatches, Command};
-
-fn build_command() -> ArgMatches {
-    Command::new(crate_name!())
-        .author(crate_authors!())
-        .version(crate_version!())
-        .about(crate_description!())
-        .arg(
-            Arg::new("Url")
-                .short('d')
-                .long("download")
-                .value_parser(value_parser!(String))
-                .help("Url to download"),
-        )
-        .arg(
-            Arg::new("Resume")
-                .long("resume")
-                .value_parser(value_parser!(String))
-                .num_args(0..=1)
-                .default_missing_value("auto")
-                .help("Resume download of the given record ID, if no ID is given the recent pending download will resume"),
-        )
-        .arg(
-            Arg::new("InputFile")
-                .short('f')
-                .long("file")
-                .value_parser(value_parser!(String))
-                .help("Filepath to the file with links"),
-        )
-        .arg(
-            Arg::new("History")
-                .long("history")
-                .action(clap::ArgAction::SetTrue)
-                .help("Display download history"),
-        )
-        .arg(
-            Arg::new("Saveas")
-                .short('s')
-                .long("save-as")
-                .help("Save file as custom name"),
-        )
-        .arg(
-            Arg::new("Threads")
-                .short('m')
-                .long("max-threads")
-                .value_parser(value_parser!(usize))
-                .help("Maximum number of concurrent downloads"),
-        )
-        .get_matches()
-}
-
-async fn url_download(args: ArgMatches, mut config: Config) -> Result<(), RawstErr> {
-    let url = args.get_one::<String>("Url").unwrap().to_string();
-
-    let save_as = args.get_one::<String>("Saveas");
-
-    let threads = args.get_one::<usize>("Threads");
+// TODO: Fuse url_download and list_download
+// TODO: Support downloading many elements from each source
 
-    // 8 threads are maximum
-    // more than 8 threads could cause rate limiting
-    let threads_limit = 1..9;
+fn first<T>(xs: Vec<T>) -> Option<T> {
```

Comment:
We used `args.get_one::<String>` a lot in here, which means that only argument was supported as I found no loops.

I used `Vec<String>` for many options, which might require some rework underneath.

Maybe I just need to make the options `String` for now?